### PR TITLE
Add CelebA-HQ dataset for benchmarking

### PR DESCRIPTION
**Pull request by bischtob**
Adds a version of the CelebA-HQ dataset for testing purposes.

## To-do
Done.

## Content
A new Datadeps.jl registration and struct for loading CelebA-HQ.

### CODE REVIEW

File: src/datasets/celeba_hq.jl [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+CelebAHQ(Tx::Type; kws...) = CelebAHQ(; Tx, kws...)
+
+function CelebAHQ(Tx::Type, resolution::Int, split::Symbol, gender::Symbol; dir=nothing)
+    DEPNAME = "CelebAHQ"
+
+    @assert split ∈ (:train, :test)
+    @assert gender ∈ (:male, :female)
+
+    if resolution in 2 .^ (0:8)
+        newsplit = split == :test ? :val : split
```
Comment (kmdeck) [inline src/datasets/celeba_hq.jl:69]: could we not use :val from the beginning? (train/test not train/val)

File: src/datasets/celeba_hq.jl [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+CelebAHQ(resolution::Symbol; kws...) = CelebAHQ(; resolution, kws...)
+CelebAHQ(Tx::Type; kws...) = CelebAHQ(; Tx, kws...)
+
+function CelebAHQ(Tx::Type, resolution::Int, split::Symbol, gender::Symbol; dir=nothing)
+    DEPNAME = "CelebAHQ"
+
+    @assert split ∈ (:train, :test)
+    @assert gender ∈ (:male, :female)
+
+    if resolution in 2 .^ (0:8)
```
Comment (kmdeck) [inline src/datasets/celeba_hq.jl:68]: we start at resolution = 32, so maybe this should be 2. ^ (5:8)?

File: src/datasets/celeba_hq.jl [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    gender::Symbol
+    features::Array{}
+end
+
+CelebAHQ(; split=:train, resolution=32, gender=:female, Tx=Float32, dir=nothing) = CelebAHQ(Tx, resolution, split, gender; dir)
+CelebAHQ(split::Symbol; kws...) = CelebAHQ(; split, kws...)
+CelebAHQ(resolution::Symbol; kws...) = CelebAHQ(; resolution, kws...)
+CelebAHQ(Tx::Type; kws...) = CelebAHQ(; Tx, kws...)
+
+function CelebAHQ(Tx::Type, resolution::Int, split::Symbol, gender::Symbol; dir=nothing)
```
Comment (kmdeck) [inline src/datasets/celeba_hq.jl:62]: what is the point of `dir`? here it is `nothing`, not sure how it is used

File: src/datasets/celeba_hq.jl (hunk 1 above)
Comment (bischtob) [reply src/datasets/celeba_hq.jl:69]: There are two different conventions. CelebA uses train and val and MLDatasets uses train and test.

File: src/datasets/celeba_hq.jl (hunk 2 above)
Comment (bischtob) [reply src/datasets/celeba_hq.jl:68]: Good catch.

File: src/datasets/celeba_hq.jl (hunk 3 above)
Comment (bischtob) [reply src/datasets/celeba_hq.jl:62]: It's conforming to MLDatasets.jl API.

**Review by kmdeck — APPROVED** [review]

**Pull request merged**